### PR TITLE
Add crop option when taking a screenshot in e2e tests

### PR DESCRIPTION
**Pull request by fluiddot**
This PR introduces a new option to crop the screenshots taken in e2e tests. Additionally, a new util has been added to take screenshots of specific elements. This will be very handy to produce screenshots of smaller areas of the editor, as well as make them more resilient to device specifics, like size, safe area, and status bar.

<details><summary>Here are some examples using the test file referenced below</summary>

| Android | iOS - iPhone | iOS - iPad |
|--------|--------|--------|
| **Title element:**<br><br>gutenberg-editor-test-visual-test-js-test-1-android | **Title element:**<br><br>gutenberg-editor-test-visual-test-js-test-1-ios | **Title element:**<br><br>gutenberg-editor-test-visual-test-js-test-1-ios |
| **Toolbar element:**<br><br>gutenberg-editor-test-visual-test-js-test-2-android | **Toolbar element:**<br><br>gutenberg-editor-test-visual-test-js-test-2-ios | **Toolbar element:**<br><br>gutenberg-editor-test-visual-test-js-test-2-ios |
| **Shortcode block:**<br><br>gutenberg-editor-test-visual-test-js-test-3-android | **Shortcode block:**<br><br>gutenberg-editor-test-visual-test-js-test-3-ios | **Shortcode block:**<br><br>gutenberg-editor-test-visual-test-js-test-3-ios |
| **Shortcode block with extra padding:**<br><br>gutenberg-editor-test-visual-test-js-test-4-android | **Shortcode block with extra padding:**<br><br>gutenberg-editor-test-visual-test-js-test-4-ios | **Shortcode block with extra padding:**<br><br>gutenberg-editor-test-visual-test-js-test-4-ios |
| **Full screen:**<br><br>gutenberg-editor-test-visual-test-js-test-5-android | **Full screen:**<br><br>gutenberg-editor-test-visual-test-js-test-5-ios | **Full screen:**<br><br>gutenberg-editor-test-visual-test-js-test-5-ios | 
| **Full screen without keyboard:**<br><br>gutenberg-editor-test-visual-test-js-test-6-android | **Full screen without keyboard:**<br><br>gutenberg-editor-test-visual-test-js-test-6-ios | **Full screen without keyboard:**<br><br>gutenberg-editor-test-visual-test-js-test-6-ios |
| **Specific position:**<br><br>gutenberg-editor-test-visual-test-js-test-8-android | **Specific position:**<br><br>gutenberg-editor-test-visual-test-js-test-8-ios | **Specific position:**<br><br>gutenberg-editor-test-visual-test-js-test-8-ios |
| **Specific height:**<br><br>gutenberg-editor-test-visual-test-js-test-9-android | **Specific height:**<br><br>gutenberg-editor-test-visual-test-js-test-9-ios | **Specific height:**<br><br>gutenberg-editor-test-visual-test-js-test-9-ios |
| **Specific width:**<br><br> | **Specific width:**<br><br> | **Specific width:**<br><br> |

</details> 

### CI jobs pass
* Observe that all tests pass, including the optional tests.

### Test new util `takeScreenshotByElement`
* Apply the following patch that incorporates a new test only for debugging.
```patch
diff --git forkSrcPrefix/__device-tests__/gutenberg-editor-test-visual.test.js forkDstPrefix/__device-tests__/gutenberg-editor-test-visual.test.js
new file mode 100644
index 0000000000000000000000000000000000000000..39bd5e02c1ab218a61c31de9ef38229a9c0ba16c
--- /dev/null
+++ forkDstPrefix/__device-tests__/gutenberg-editor-test-visual.test.js
@@ -0,0 +1,65 @@
+/**
+ * Internal dependencies
+ */
+const { blockNames } = editorPage;
+import { takeScreenshot, takeScreenshotByElement } from './utils';
+
+it( 'test', async () => {
+	// Add Shortcode block
+	await editorPage.addNewBlock( blockNames.shortcode );
+	const shortcodeBlock = await editorPage.getBlockAtPosition(
+		blockNames.shortcode
+	);
+	await shortcodeBlock.click();
+	await editorPage.driver.sleep( 500 );
+	// Take screenshots:
+	// - Title
+	expect(
+		await takeScreenshotByElement( await editorPage.getTitleElement() )
+	).toMatchImageSnapshot();
+	// - Toolbar
+	expect(
+		await takeScreenshotByElement( await editorPage.getToolbar() )
+	).toMatchImageSnapshot();
+	// - Shortcode block
+	expect(
+		await takeScreenshotByElement( shortcodeBlock )
+	).toMatchImageSnapshot();
+	// - Shortcode block with extra padding
+	expect(
+		await takeScreenshotByElement( shortcodeBlock, {
+			padding: 10,
+		} )
+	).toMatchImageSnapshot();
+	// - Full screen
+	expect( await takeScreenshot() ).toMatchImageSnapshot();
+	// - Full screen without keyboard
+	expect(
+		await takeScreenshot( {
+			withoutKeyboard: true,
+		} )
+	).toMatchImageSnapshot();
+	// - Specific position
+	expect(
+		await takeScreenshot( {
+			crop: {
+				x: 0,
+				y: 200,
+				width: 300,
+				height: 300,
+			},
+		} )
+	).toMatchImageSnapshot();
+	// - Specific height
+	expect(
+		await takeScreenshot( {
+			crop: { x: 0, y: 0, height: 300 },
+		} )
+	).toMatchImageSnapshot();
+	// - Specific width
+	expect(
+		await takeScreenshot( {
+			crop: { width: 200 },
+		} )
+	).toMatchImageSnapshot();
+} );
``` 
* Run e2e tests on both platforms and check that generated screenshots match the expectations (see inline comments added to each `expect` statement).

PR submission checklist:

- [ ] I have considered adding unit tests where possible.
- [ ] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

File: __device-tests__/utils.js [hunk 1]
```diff
@@ -5,19 +5,72 @@ import jimp from 'jimp';
...
+ * @param {number}  [options.offset.width]     Width offset to crop.
+ * @param {number}  [options.offset.height]    Height offset to crop.
  * @return {Buffer} Sreenshot image.
  */
 export async function takeScreenshot(
-	options = { withoutKeyboard: false, heightPercentage: undefined }
+	options = {
+		withoutKeyboard: false,
+		heightPercentage: undefined,
+		crop: undefined,
```
Comment (dcalhoun) [inline __device-tests__/utils.js]: I believe it is unnecessary to set `undefined`, that is the inherent value. 
```suggestion
```

File: __device-tests__/utils.js [hunk 2]
```diff
@@ -5,19 +5,72 @@ import jimp from 'jimp';
...
  * Helper to take a screenshot and manipulate it.
  *
  * @param {Object}  options                    Options
  * @param {boolean} [options.withoutKeyboard]  Prevents showing the keyboard in the screenshot.
  * @param {number}  [options.heightPercentage] Specify a custom height in percentage.
+ * @param {Object}  [options.crop]             Specify values to crop the screenshot.
+ * @param {number}  [options.offset.x]         X offset to crop.
+ * @param {number}  [options.offset.y]         Y offset to crop.
+ * @param {number}  [options.offset.width]     Width offset to crop.
+ * @param {number}  [options.offset.height]    Height offset to crop.
```
Comment (dcalhoun) [inline __device-tests__/utils.js]: I do not see any references to `offset`, should this be `crop` instead?

```suggestion
 * @param {number}  [options.crop.x]         X offset to crop.
 * @param {number}  [options.crop.y]         Y offset to crop.
 * @param {number}  [options.crop.width]     Width offset to crop.
 * @param {number}  [options.crop.height]    Height offset to crop.
```

File: __device-tests__/utils.js [hunk 3]
```diff
@@ -5,19 +5,72 @@ import jimp from 'jimp';
...
+	}
+	return takeScreenshot( { crop } );
+}
+
 /**
  * Helper to take a screenshot and manipulate it.
  *
  * @param {Object}  options                    Options
  * @param {boolean} [options.withoutKeyboard]  Prevents showing the keyboard in the screenshot.
  * @param {number}  [options.heightPercentage] Specify a custom height in percentage.
```
Comment (dcalhoun) [inline __device-tests__/utils.js]: I was considering the fact that the new `crop` option is a more robust solution for the existing `heightPercentage`. I wondered if we might replace `heightPercentage` with `crop` usage, potentially composed in some manner to make it just as approachable and simple as `heightPercentage`. 

However, after researching `heightPercentage`'s origin and current usage, I noted we [removed its one usage](https://github.com/wordpress-mobile/gutenberg-mobile/commit/1c3f9c39587ddcd6dc7b8354caba51d36c7e4c34#diff-aa9661e5ba881a583b0167cdafcd09388dbbc4948253b0a176d24bad671f89c3L206). What do you think about removing `heightPercentage` entirely and considering it now irrelevant thanks to the new `crop` option?

**Review by dcalhoun — COMMENTED** [review]
This is fantastic! 👏🏻 Thank for creating such a useful utility. I can see this simplifying many tests and reducing the storage size of our visual snapshots.

File: __device-tests__/utils.js (hunk 2 above)
Comment (fluiddot) [reply __device-tests__/utils.js]: Yep, I originally named the option `offset` and then changed it to `crop`, but I forgot to update this part. I'll update it.

File: __device-tests__/utils.js (hunk 3 above)
Comment (fluiddot) [reply __device-tests__/utils.js]: > However, after researching `heightPercentage`'s origin and current usage, I noted we [removed its one usage](https://github.com/wordpress-mobile/gutenberg-mobile/commit/1c3f9c39587ddcd6dc7b8354caba51d36c7e4c34#diff-aa9661e5ba881a583b0167cdafcd09388dbbc4948253b0a176d24bad671f89c3L206). What do you think about removing `heightPercentage` entirely and considering it now irrelevant thanks to the new `crop` option?

Ah, good catch. Definitely, if we no longer use `heightPercentage` I'll be happy to remove this part. Thanks for pointing it out 🙇 !

File: __device-tests__/utils.js (hunk 2 above)
Comment (fluiddot) [reply __device-tests__/utils.js]: I fixed this in https://github.com/wordpress-mobile/gutenberg-mobile/pull/6038/commits/a9885842f0899fb4d64a03bff735486f08b1786a, although I changed a bit the structure in https://github.com/wordpress-mobile/gutenberg-mobile/pull/6038/commits/c718739bfed8508e39b0c168d968894928741fdc

File: __device-tests__/utils.js (hunk 1 above)
Comment (fluiddot) [reply __device-tests__/utils.js]: Suggestion applied in https://github.com/wordpress-mobile/gutenberg-mobile/pull/6038/commits/c718739bfed8508e39b0c168d968894928741fdc.

File: __device-tests__/utils.js (hunk 3 above)
Comment (fluiddot) [reply __device-tests__/utils.js]: Suggestion applied in https://github.com/wordpress-mobile/gutenberg-mobile/pull/6038/commits/831cc2b48ef190d6fe17f0018aa935a64dbff131.

**Review by dcalhoun — APPROVED** [review]
This is awesome! 🚀

**Pull request merged**